src/filesystem/nodes/mod.rs

```diff
@@ -495,4 +495,100 @@ pub(crate) mod test {
         let _mut_data = test_node.data_mut().await;
         assert!(test_node.cid.is_dirty().await);
     }
+
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test(async))]
+    #[cfg_attr(not(target_arch = "wasm32"), tokio::test)]
+    async fn test_empty_file_round_trip() {
+        let mut rng = ChaCha20Rng::from_entropy();
+        let test_node = setup_test_node(&mut rng, NodeData::empty_file()).await;
+
+        let mut buffer = Vec::new();
+        test_node.encode(&mut buffer).await.unwrap();
+        #[rustfmt::skip]
+        assert_eq!(
+            buffer,
+            &[
+                // cid (32 bytes)
+                123, 213, 174, 5, 166, 228, 27, 101, 174, 250, 234, 210, 62, 78, 210, 81, 217, 42, 190, 46, 153, 237, 109, 11, 254, 212, 23, 217, 223, 1, 12, 205,
+                // node_data_len (4 bytes)
+                88, 0, 0, 0,
+                // permanent_id (8 bytes)
+                2, 2, 2, 2, 2, 2, 2, 2,
+                // vector_clock (8 bytes)
+                0, 0, 0, 0, 0, 0, 0, 0,
+                // parent_present flag (1 byte) + parent_id (8 bytes)
+                1, 1, 1, 1, 1, 1, 1, 1, 1,
+                // owner_id (fingerprint; 32 bytes)
+                4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,  4, 4, 4, 4,
+                // created_at (8 bytes)
+                210, 2, 150, 73, 0, 0, 0, 0,
+                // modified_at (8 bytes)
+                211, 2, 150, 73, 0, 0, 0, 0,
+                // name_type (1 byte) + name_length (1  byte) + name (7 bytes)
+                1, 7, 84, 101, 115, 116, 68, 105, 114,
+                // metadata entries len (1 byte) + permission flag (1 byte) + children count (2 byte)
+                0, 0, 0, 0,
+                // content len (1 byte) + file type (1 byte)
+                0, 4
+            ]
+        );
+        let (remaining, parsed) = Node::parse(Stream::new(&buffer), test_node.id).unwrap();
+        let remaining: Vec<u8> = remaining.to_vec();
+        assert_eq!(Vec::<u8>::new(), remaining);
+        assert_node_equality(&test_node, &parsed);
+    }
```

Comment:
Great test! Hahah I didn't mean to go quite this deep but its a very useful one. Appreciate it

Reply:
Yeah that one was more for my benefit and I imagine someone new to the lib to quickly get an idea of roughly what's happening